sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -30,6 +31,13 @@ public HttpClientTransport() : this(CreateDefaultClient())
         {
         }
 
+        /// <summary>
+        /// Creates a new <see cref="HttpClientTransport"/> instance using default configuration.
+        /// </summary>
+        /// <param name="options">The <see cref="HttpClientTransportOptions"/> that to configure the behavior of the transport.</param>
+        public static HttpClientTransport CreateWithDefaultSettings(HttpClientTransportOptions? options = null)
```

Comment:
I think this should be a `HttpPipelineTransport.Create()` so we can create HttpWebRequest or HttpClientTransport based on the environment and settings.

Reply:
Does that imply that we also need `HttpPipelineTransportOptions` as a base abstract as well?

Reply:
I think we need HttpPipelineTransportOptions as the only option class. HttpPipelineTransportOptions would represent a set of features that Azure.Core transports provide and clients shouldn't know which exact transport is being created as long as it implements the required options correctly.

Reply:
> tpPipelineTransportOptions as the only option class

Yes, that makes sense.


> I think this should be a `HttpPipelineTransport.Create()` so we can create HttpWebRequest or HttpClientTransport based on the environment and settings.

In order to have the transport type selection happen naturally based on environment and settings, it seems like we'd need to expose the transport options in `ClientOptions` somehow, since that is where it currently happens as part of `DefaultClientOptions`.